Source/ARTPushActivationState.m

```diff
@@ -117,6 +117,9 @@ - (ARTPushActivationState *)transition:(ARTPushActivationEvent *)event {
     }
     else if ([event isKindOfClass:[ARTPushActivationEventCalledActivate class]]) {
         [self.machine registerForAPNS];
+#if TARGET_OS_IOS
+        [self.machine.rest setupLocalDevice];
```

Comment:
This order doesn't match that of RSH3a2b and RSH3a2d.

Reply:
https://github.com/ably/ably-cocoa/pull/1847/commits/9277ab083ae1fc5aec060ba98632a4083986ce93

Reply:
Unresolved due to code changed - https://github.com/ably/ably-cocoa/pull/1847/commits/7159d42f255b9c593bc0f94f0b32eb2c57afee17

Reply:
Would you mind explaining why this code moved into `validateAndSync`, please?

Reply:
This is to keep a proper call order `setupLocalDevice_nosync` - `registerForAPNS` and also call `setupLocalDevice_nosync` after `61002` error check. That's why I wanted that return value, turned out it can be solved by moving everything inside `validateAndSync`. Simple and doesn't break anything, why not then?